### PR TITLE
change GPIO18 access rights for homegear

### PR DESCRIPTION
**Pull request by mstormi**
Signed-off-by: Markus Storm <markus.storm@gmx.net>

### CODE REVIEW

File: functions/system.bash [hunk 1]
```diff
@@ -315,6 +314,16 @@ permissions_corrections() {
...
     chown homegear:homegear -R /var/log/homegear
+
+    # homeMatic/homegear controller HM-MOD-RPI-PCB uses GPIO 18 to reset HW
+    if [[ ! -d "${gpioDir}/gpio18" ]]; then
+      echo "18" > /sys/class/gpio/export
+      echo "out" > /sys/class/gpio/gpio/direction
+      echo "0" > /sys/class/gpio/gpio/value
+    fi
+    if [[ ! -d "${gpioDir}/gpio18" ]]; then echo "FAILED (set GPIO 18 access)"; return 1; fi
+    chgrp gpio ${gpioDir}/gpio18/*
```
[the remark this answers is not in the document]
Comment (mstormi) [reply functions/system.bash:325]: L324 does. We shouldn't overdo and check every single atomic command. Why wouldn't it execute (we run as root) ?

File: functions/system.bash [hunk 2]
```diff
@@ -275,9 +275,8 @@ srv_bind_mounts() {
 ##    permissions_corrections()
 ##
 permissions_corrections() {
-  local openhabFolders
-
-  openhabFolders=(/etc/openhab2 /var/lib/openhab2 /var/log/openhab2 /usr/share/openhab2)
+  local openhabFolders=(/etc/openhab2 /var/lib/openhab2 /var/log/openhab2 /usr/share/openhab2)
+  local gpioDir=/sys/devices/platform/soc
```
[the remark this answers is not in the document]
Comment (mstormi) [reply functions/system.bash]: ok

File: functions/system.bash [hunk 3]
```diff
@@ -315,6 +314,16 @@ permissions_corrections() {
     chown homegear:homegear /etc/homegear/rpcclients.conf
     chmod 400 /etc/homegear/rpcclients.conf
     chown homegear:homegear -R /var/log/homegear
+
+    # homeMatic/homegear controller HM-MOD-RPI-PCB uses GPIO 18 to reset HW
+    if [[ ! -d "${gpioDir}/gpio18" ]]; then
```
[the remark this answers is not in the document]
Comment (mstormi) [reply functions/system.bash]: ok

File: functions/system.bash [hunk 4]
```diff
@@ -315,6 +314,16 @@ permissions_corrections() {
...
+      echo "18" > /sys/class/gpio/export
+      echo "out" > /sys/class/gpio/gpio/direction
+      echo "0" > /sys/class/gpio/gpio/value
+    fi
+    if [[ ! -d "${gpioDir}/gpio18" ]]; then echo "FAILED (set GPIO 18 access)"; return 1; fi
```
[the remark this answers is not in the document]
Comment (mstormi) [reply functions/system.bash]: ok

File: functions/system.bash (hunk 1 above)
[the remark this answers is not in the document]
Comment (ecdye) [reply functions/system.bash:325]: I did not notice that there was a check on L324. Somehow I missed that sorry.

**Pull request merged**